common/src/main/java/com/wynntils/mc/mixin/GuiMixin.java

```diff
@@ -167,4 +170,33 @@ private void onRenderHeartsPre(
             ci.cancel();
         }
     }
+
+    @Inject(
+            method = "Lnet/minecraft/client/gui/Gui;renderSelectedItemName(Lnet/minecraft/client/gui/GuiGraphics;)V",
+            at = @At("HEAD"),
+            cancellable = true)
+    private void onRenderSelectedItemNamePre(GuiGraphics guiGraphics, CallbackInfo ci) {
+        if (!MixinHelper.onWynncraft()) return;
+
+        RenderEvent.Pre event =
+                new RenderEvent.Pre(guiGraphics, 0, this.minecraft.getWindow(), RenderEvent.ElementType.HELD_ITEM_NAME);
+        MixinHelper.post(event);
+
+        if (event.isCanceled()) {
+            ci.cancel();
+        }
+    }
+
+    @ModifyVariable(
+            method = "Lnet/minecraft/client/gui/Gui;setOverlayMessage(Lnet/minecraft/network/chat/Component;Z)V",
+            at = @At("HEAD"),
+            ordinal = 0)
+    private Component setActionBarMessage(Component value) {
+        if (!MixinHelper.onWynncraft()) return value;
+
+        ActionBarMessageSetEvent event = new ActionBarMessageSetEvent(value);
+        MixinHelper.post(event);
+
+        return event.getMessage();
+    }
```

Comment:
In 1.21 there is a `renderOverlayMessage` so this can be replaced with that

Reply:
I think you should try Forge. I am at least 90% sure this mixin won't work. Forge has a separate `ForgeGui` class on this version (and patches the regular `Gui` class on 1.21`. This will be a bit of a pain to port, but oh well, this is the reason I've separated Fabric/Forge only mixins in 1.21.

Reply:
Ah, I mainly wanted to start doing the "vanilla overlays" so I didn't have to port removehud again so if it's easier it can be done after 1.21, it's not necessarily needed for 1.20

Reply:
Action bar works for forge but not held item, for some reason they have a parameter to shift the y level in their `Gui` class `renderSelectedItemName(GuiGraphics arg, int yShift)` 

It's not in `ForgeGui` so I don't think it can be injected in `ForgeGuiMixin` unless I'm just doing the mixin wrong. So definitely probably better saving this for 1.21 and if we do want to refactor overlays then maybe that first too